### PR TITLE
chore(state): extend error handling in core accessor

### PR DESCRIPTION
**Pull request by vgonkivs**
## Overview

Extend error handling in core accessor

## Checklist

- [ ] New and updated code has appropriate documentation
- [ ] New and updated code has new and/or updated testing
- [ ] Required CI checks are passing
- [ ] Visual proof for any user facing features like CLI or documentation updates
- [ ] Linked issues closed with keywords

### CODE REVIEW

File: blob/service.go [hunk 1]
```diff
@@ -70,7 +70,7 @@ func (s *Service) Submit(ctx context.Context, blobs []*Blob) (uint64, error) {
 	if err != nil {
 		return 0, err
 	}
-	return uint64(resp.Height), nil
+	return uint64(resp.Height), err
```
Comment (Wondertan) [inline blob/service.go]: The error is always nil here

File: state/helpers.go [hunk 2]
```diff
@@ -1,21 +1,64 @@
...
 		return status.Error(codes.Unauthenticated, resp.Log)
-	case sdk_errors.ErrKeyNotFound.ABCICode():
+	case sdk_errors.ErrKeyNotFound:
 		return status.Error(codes.NotFound, resp.Log)
 	default:
 		return status.Error(codes.Unknown, resp.Log)
 	}
 }
+
+func codeToError(code uint32) error {
```
Comment (Wondertan) [inline state/helpers.go:25]: Use https://github.com/cosmos/cosmos-sdk/blob/main/errors/errors.go#L75 instead

**Review by Wondertan — CHANGES_REQUESTED** [review]

File: state/core_access.go [hunk 3]
```diff
@@ -177,6 +178,10 @@ func (ca *CoreAccessor) SubmitPayForBlob(
 		ca.lastPayForBlob = time.Now().UnixMilli()
 		ca.payForBlobCount++
 	}
+
+	if response.Code != 0 {
```
Comment (Wondertan) [inline state/core_access.go:182]: very optional nit: can be coupled with the if above

**Review by Wondertan — APPROVED** [review]

File: state/core_access.go (hunk 3 above)
Comment (vgonkivs) [reply state/core_access.go:182]: I would suggest keeping it separately: the previous statement contains metrics handling, and the new one is error handling.

**Review by walldiss — APPROVED** [review]

**Pull request merged**